File: backend/ticket-service/cqrs/handler/QueryHandler.py
class QueryHandler:
    def __init__(self):
        self.handlers = {}

    def register(self, query_type, handler):
        self.handlers[query_type] = handler
        print(f"[QueryHandler] Registered handler for: {query_type}")

    async def dispatch(self, query_type, query):
        handler = self.handlers.get(query_type)
        if not handler:
            error = f"No handler registered for query: {query_type}"
            print(f"[QueryHandler] {error}")
            raise Exception(error)
        try:
            print(f"[QueryHandler] Dispatching query: {query_type} {query}")
            result = await handler.execute(query)
            print(f"[QueryHandler] Success: {query_type} {result}")
            return result
        except Exception as error:
            print(f"[QueryHandler] Fail: {query_type} {error} {query}")
            raise error

    def execute(self, *args, **kwargs):
        raise NotImplementedError("execute metodu override edilmeli.")

# Global instance
query_handler = QueryHandler()

File: backend/auth-service/src/cqrs/handler/query.handler.js
import logger from '../../config/logger.js';
import translation from '../../config/translation.js';

export class QueryHandler {
  constructor() {
    this.handlers = new Map();
  }

  register(queryType, handler) {
    this.handlers.set(queryType, handler);
    logger.info(translation('cqrs.handler.query.logs.registered'), { queryType });
  }

  async dispatch(queryType, query) {
    const handler = this.handlers.get(queryType);
    
    if (!handler) {
      const error = `No handler registered for query: ${queryType}`;
      logger.error(translation('cqrs.handler.query.logs.noHandler'), { queryType });
      throw new Error(translation('cqrs.handler.query.logs.noHandler', { queryType }));
    }

    try {
      logger.info(translation('cqrs.handler.query.logs.dispatching'), { queryType, query });
      const result = await handler.execute(query);
      logger.info(translation('cqrs.handler.query.logs.success'), { queryType, result });
      return result;
    } catch (error) {
      logger.error(translation('cqrs.handler.query.logs.fail'), { queryType, error, query });
      throw error;
    }
  }
}

export const queryHandler = new QueryHandler(); 